cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaMetadataStateManagerMockTest.java

```diff
@@ -0,0 +1,609 @@
+                new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME),
+                vertx,
+                CONFIG,
+                supplier,
+                new PlatformFeaturesAvailability(false, KUBERNETES_VERSION),
+                kafka,
+                List.of(BROKERS, CONTROLLERS),
+                createKafkaCluster(KAFKA),
+                CLUSTER_CA,
+                CLIENTS_CA);
+
+        KafkaStatus status = new KafkaStatus();
+
+        Checkpoint async = context.checkpoint();
+        kr.reconcile(status, Clock.systemUTC()).onComplete(res -> context.verify(() -> {
+            assertTrue(status.getConditions().stream().anyMatch(condition -> "KafkaMetadataStateWarning".equals(condition.getReason())));
+            assertEquals(status.getConditions().get(2).getMessage(), "The strimzi.io/kraft annotation can't be set to rollback during a migration process." +
+                    "It can be used in post migration to start rollback process.");
+            assertEquals(status.getKafkaMetadataState(), KRaftMigration.name());
+            async.flag();
+        }));
+    }
+
+    /** Mock for KafkaMetadataStateManager class */
+    static class MockKafkaMetadataStateManager extends KafkaMetadataStateManager {
+
+        /**
+         * Constructor
+         *
+         * @param reconciliation             Reconciliation information
+         * @param kafkaCr                    instance of the Kafka CR
+         * @param useKRaftFeatureGateEnabled if the UseKRaft feature gate is enabled on the operator
+         */
+        public MockKafkaMetadataStateManager(Reconciliation reconciliation, Kafka kafkaCr, boolean useKRaftFeatureGateEnabled) {
+            super(reconciliation, kafkaCr, useKRaftFeatureGateEnabled);
+        }
+
+        @Override
+        protected KafkaAgentClient initKafkaAgentClient(Reconciliation reconciliation, Secret clusterCaCertSecret, Secret coKeySecret) {
+            KafkaAgentClient agentClient = mock(KafkaAgentClient.class);
+            when(agentClient.getKRaftMigrationState(any())).thenReturn(new KRaftMigrationState(2), new KRaftMigrationState(1));
+            return agentClient;
+        }
+    }
+
+    /** Mock for KafkaReconciler class */
+    static class MockKafkaReconciler extends KafkaReconciler {
+        private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(MockKafkaReconciler.class.getName());
+
+        public MockKafkaReconciler(Reconciliation reconciliation, Vertx vertx, ClusterOperatorConfig config, ResourceOperatorSupplier supplier, PlatformFeaturesAvailability pfa, Kafka kafka, List<KafkaNodePool> nodePools, KafkaCluster kafkaCluster, ClusterCa clusterCa, ClientsCa clientsCa) {
+            super(reconciliation, kafka, nodePools, createKafkaCluster(kafka), clusterCa, clientsCa, config, supplier, pfa, vertx, new MockKafkaMetadataStateManager(reconciliation, kafka, config.featureGates().useKRaftEnabled()));
```

Comment:
I think the `MockKafkaMetadataStateManager` should be passed to the ctor on line 592 and then here.